Adding active particle docs (#85)
enzo-project/enzo-dev: doc/manual/source/parameters/index.rst

``MagneticSupernovaDuration`` (external)
    The duration (in years) over which the total magnetic supernova energy is injected. This should be set to at least 5 times the minimum timestep of the simulation. Default: 5e4

.. _active_particles_parameters:

Active Particles
~~~~~~~~~~~~~~~~

To allow for the creation of an active particle the following line must be added to the parameter file or restart file:

> ``AppendActiveParticleType = <ActiveParticleType``

Reviewer: ```suggestion
``AppendActiveParticleType = <ActiveParticleType>``
```

See :ref:`active_particles` for the types of active particles that exist.

SmartStar Parameters
~~~~~~~~~~~~~~~~~~~~

SmartStar Accretion

> ^^^^^^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
^^^^^^^^^^^^^^^^^^^
```

``SmartStarAccretion`` (external)

Set to 1 to turn on spherical Bondi-Hoyle accretion based on the formalisms

> presented in Krumholz et al. (2004) - https://iopscience.iop.org/article/10.1086/421935/pdf

Reviewer: ```suggestion
presented in `Krumholz et al. (2004) <https://iopscience.iop.org/article/10.1086/421935/pdf>`__
```

Set to 4 to turn on angular momentum limited accretion based on the formalism adopted

> by Rosas-Guevara et al. (2015) - https://arxiv.org/pdf/1312.0598.pdf

Reviewer: ```suggestion
by `Rosas-Guevara et al. (2015) <https://arxiv.org/pdf/1312.0598.pdf>`__.
```

Set to 5 to again turn on spherical Bondi-Hoyle accretion but this time the
formalism includes a correction which accounts for the vorticity of the gas

> based on Krumholz et al. (2005) - https://iopscience.iop.org/article/10.1086/426051/pdf

Reviewer: ```suggestion
based on `Krumholz et al. (2005) <https://iopscience.iop.org/article/10.1086/426051/pdf>`__.
```

> Set to 6 to use the viscous angular momentum prescription from DeBuhr et al. (2010)

Reviewer: ```suggestion
Set to 6 to use the viscous angular momentum prescription from `DeBuhr et al. (2010) <https://arxiv.org/pdf/0909.2872.pdf>`__.
```

> - https://arxiv.org/pdf/0909.2872.pdf

Reviewer: ```suggestion
```

  
Set to 7 to use a  modified accretion rate based on the alpha-disk model of Shakura & Sunyaev (1973)

> This scheme is based on Cen et al. https://iopscience.iop.org/article/10.1088/0004-637X/755/1/28/pdf

Reviewer: ```suggestion
This scheme is based on `Cen et al. <https://iopscience.iop.org/article/10.1088/0004-637X/755/1/28/pdf>`__.
```

Set to 8 to employ the converging mass flux approach. This is not based on any indirect properties of the
gas as above and simply measures the flow of gas through the accetion radius of the ``SmartStar``. It is based

> on the accretion mechanism used by Bleuler in Ramses https://arxiv.org/pdf/1409.6528.pdf and in the Enzo simulations

Reviewer: ```suggestion
on the accretion mechanism used by `Bleuler <https://arxiv.org/pdf/1409.6528.pdf>`__  in Ramses and in Enzo
```

> by Regan et al. (2018, 2019) https://arxiv.org/pdf/1803.04527.pdf, https://arxiv.org/pdf/1811.04953.pdf

Reviewer: ```suggestion
by Regan et al. (`2018 <https://arxiv.org/pdf/1803.04527.pdf>`__, `2019 <https://arxiv.org/pdf/1811.04953.pdf>`__).
```

Default: 8

> Accretion is never capped by default but can capped at the Eddington rate as notes below. 

Reviewer: ```suggestion
Accretion is never capped by default but can be capped at the Eddington rate as notes below. 
```

   

SmartStar Feedback

> ^^^^^^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
^^^^^^^^^^^^^^^^^^
```

    The stellar radiative feedback is divided up into 5 energy bins. The energy bins have energies of 2.0 eV, 12.8 eV, 14.0 eV, 25.0 eV
    and 200 eV. The fraction of energy assigned to each bin is determined using the PopIII tables from Schaerer et. al 2002 Table 4.
    The spectrum for a PopIII star and SMS are different. For a PopIII star a spectrum for a 40 Msolar star is assumed and
    weighted accordingly. For a SMS a 1000 Msolar star is assumed and weighted accordingly.
    Future improvements to the SEDs employed here are under active investigation. 
    Default: 0

``SmartStarBHFeedback`` (external)
    This is a master switch on black hole feedback. Must be turned on if you want black hole feedback. Default: 0

``SmartStarBHRadiativeFeedback`` (external)
    This parameter controls whether black hole radiative feedback gets turned on or not. When turned on the radiative
    feedback from a black hole depends both on the mass of the black hole and the accretion rate onto the black hole. Both of these
    quantities are captured and stored as part of the ``SmartStar``. Details of the SED used can be found in the appendix of

>     https://arxiv.org/abs/1811.04953 and is made from assuming a multi-colour disk for the accretion disk and a corona fit by a

Reviewer: ```suggestion
    `Regan et al. 2019 <https://arxiv.org/abs/1811.04953>`__ and is made from assuming a multi-colour disk for the accretion disk and a corona fit by a
```

    power law.  The radiation emitted by the accretion disk is hard-coded is be emitted by 5 bins with energies of
    2.0 eV, 12.8 eV, 19.1 eV, 217.3 eV and 5190 eV. The fraction of energy assigned to each bin is then determined by the mass of the

>     black hole and the associated accretion rate at a given time. The formulism is valid for black hole masses between 1 Msolar and

Reviewer: ```suggestion
    black hole and the associated accretion rate at a given time. The formalism is valid for black hole masses between 1 Msolar and
```

    1e9 Msolar and for accretion rates between 1e-6 Msolar/yr and 1e3 Msolar/yr.  Default: 0

``SmartStarBHThermalFeedback`` (external)
    This parameter controls whether the black hole thermal feedback gets turned. Thre thermal energy is generated by feedback through the
    accretion process. If this is turned on then the ``SmartStarBHRadiativeFeedback`` should presumably be turned off unless you have a
    good reason to include both. The efficiency, epsilon, depends on both the spin of the black hole and the ISCO oribit. In order to
    calculate this accurately Eqn 32 from Abromowicz & Fragile (2013) is used. See ActiveParticle_SmartStar.h. The feedback is released
    iostropically in a sphere surrounding the SmartStar particle.  Default: 0

``SmartStarBHJetFeedback`` (external)

>     The methodology for this algorithm is based on that of Kim et. al (2011) (https://arxiv.org/pdf/1106.4007.pdf). Jets can be activiated

Reviewer: ```suggestion
    The methodology for this algorithm is based on that of `Kim et. al (2011) <https://arxiv.org/pdf/1106.4007.pdf>`__. Jets can be activated
```

    when a spinning black hole is accreting. The jets are bipolar and are set along the angular momentum vector of the SmartStar. The
    velocity of the jet(s) is set by a separate parameter below. No other parameters need to be set to activate the jet.  Default: 0

``SmartStarEddingtonCap`` (external)

>     This parameter allows for accretion onto the SmartStar to be capped at the Eddington limit. I see no good physical reason for doing this in

Reviewer: ```suggestion
    This parameter allows for accretion onto the SmartStar to be capped at the Eddington limit.
```

>     general.  Default: 0

Reviewer: ```suggestion
    Default: 0
```

``SmartStarSpin`` (external)

>     The dimensionless spin of the SmartStar particle. This is a very uncontrained parameter and cannot be readily computed on the fly. This parameter

Reviewer: ```suggestion
    The dimensionless spin of the SmartStar particle. This is a very unconstrained parameter and cannot be readily computed on the fly. This parameter
```

    should be set if you want to have jet feedback. Setting this is zero and turning on jet feedback wouldn't make sense. The default is set to be 0.7 and
    this is probably reasonable. 
    Default: 0.7

``SmartStarSMSLifetime`` (external)

>     This is the lifetime for a supermassive star. After this time has elapsed a SmartStar particle which is behaving like a SMS will collapase

Reviewer: ```suggestion
    This is the lifetime for a supermassive star in years. After this time has elapsed a SmartStar particle which is behaving like a SMS will collapse
```

    directly into a black hole with no supernova event. Default: 1e6

``SmartStarJetVelocity`` (external)
    The velocity that the jets are ejected at. Typically jets are observed to travel at a substantial fraction of the speed of light -
    especially those ejected during periods of high accretion. However,

>     as mass gets entrained on the jet it slows down. The units of this parameter are as a fraction of the speed of light. Default: 0.1

Reviewer: ```suggestion
    as mass gets entrained in the jet it slows down. The units of this parameter are as a fraction of the speed of light. Default: 0.1
```

``SmartStarFeedbackJetsThresholdMass`` (external)

>     Jets are only ejected once this amount of mass is available for ejected after an accretion event. Therefore, if there is very limited

Reviewer: ```suggestion
    Jets are only ejected once this amount of mass is available for ejection after an accretion event. Therefore, if there is very limited
```

    accretion and this parameter is set high then jets will be very infrequent. In units of solar masses. Default: 1.0

``SmartStarSuperEddingtonAdjustment`` (external)

>     As accretion rates exceed the canonical Eddington rate the radiative efficiency of the feedback changes. We use the fits from Madau et al. (https://arxiv.org/pdf/1402.6995.pdf)

Reviewer: ```suggestion
    As accretion rates exceed the canonical Eddington rate the radiative efficiency of the feedback changes. We use the fits from `Madau et al. <https://arxiv.org/pdf/1402.6995.pdf>`__
```

>     to adjust the efficiency when accretion enters the super-critical regime. The fits are based on the slim-disk model of accretion which generate inefficient feedback.

Reviewer: ```suggestion
    to adjust the efficiency when accretion enters the super-critical regime. The fits are based on the slim-disk model of accretion which generates inefficient feedback.
```
